### DISCUSSION TITLE
Mixing triplets in one folder

### DISCUSSION THREAD

**Discussion by srebrnyp** [microsoft/vcpkg#30310] [Q&A] [answered]
I want to mix triplets into one folder under manifest mode.
For example i have a project where i need several libraries in static mode and some other dynamic.
Let's say grpc to be static and openssl to be dynamic.
Is it possible somehow in single manifest file?
When i try to invoke two times installation with differ triplet **on the same folder**, the second invoke of install just removes what the first installation does. (maybe there is an option to not remove previous installations?)

I can only compile this libraries separately with differ triplets into differ folders and after that use script to move installations into single folder. But this doesn't looks pretty to be honest.
Something like that:
static vcpkg.json configuration file is located in 'static_conf ' subfolder with grpc lib.
static_conf/vcpkg.json  [grpc]:
```
{
    "name": "...",
    "version": "...",
    "builtin-baseline": "...",
    "dependencies": [
        grpc
    ]
}
```
vpkg.json [openssl]:
```
{
    "name": "...",
    "version": "...",
    "builtin-baseline": "...",
    "dependencies": [
        openssl
    ]
}
```

```
${VCPKG_EXE} install --triplet=x64-linux --x-manifest-root=static_conf --x-install-root=static
${VCPKG_EXE} install --triplet=x64-linux-dynamic --x-install-root=dynamic

mkdir -p install/release
mkdir -p install/debug
...
copy all libraries from static/dynamic triplets into install folder
```

**Comment by autoantwort** [comment, marked as answer]
You can create your own triplet: https://learn.microsoft.com/en-us/vcpkg/users/triplets#per-port-customization
